include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -305,7 +305,7 @@ struct __parallel_transform_reduce_impl
     {
         using _NoOpFunctor = unseq_backend::walk_n<oneapi::dpl::identity>;
         using _ReduceKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
-            __reduce_kernel, _CustomName, _ReduceOp, _TransformOp, _NoOpFunctor, _Ranges...>;
+            __reduce_kernel, _CustomName, _Tp, _ReduceOp, _TransformOp, _NoOpFunctor, _Ranges...>;
```

Comment:
As mentioned offline, I think that we may want to consider adding `_Commutative` and `_VecSize` to the list, even if it is not explicitly required based upon the current usage of `__parallel_transform_reduce_impl`. 

Perhaps there is some bigger downside to further growing this type list, but it seems that we are laying a trap for future possible usages of this struct as is.